frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -17,117 +17,140 @@ import CustomTextArea from '@Commons/Input/TextArea/CustomTextArea';
 import { faWheatAwn } from '@fortawesome/free-solid-svg-icons';
 import initialFarmDetails from '@Constants/InitialFarmDetails';
 import FieldDetailInterface from 'src/Interface/FieldDetailsInterface';
-import {
-  StyledFarmInfo,
-  StyledTextAreaContainer,
-  StyledButtonGroupContainer,
-  StyledListContainer,
-  StyledListItem,
-} from './FieldsAndSoil.style';
-
-const initialFieldValues = initialFarmDetails.Fields[0];
+import FieldsButtonComponent from './FieldsButtonComponent';
+import FieldsListComponent from './FieldsListComponent';
+import { StyledFarmInfo, StyledTextAreaContainer } from './FieldsAndSoil.style';
+import { StyledButtonGroupContainer } from './FieldsButtonComponent.styles';
 
 const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updateFarmDetails }) => {
-  const [fieldsInfo, setFieldsInfo] = useState(farmDetails);
+  // Builds field info inside the field form module.
+  const [, setFieldsInfo] = useState(farmDetails);
+  const [fieldIndex, setFieldIndex] = useState(0);
+  const [initialFieldValues, setInitialFieldValues] = useState(
+    initialFarmDetails.Fields[fieldIndex],
+  );
+  // Only triggered once, it would show list and persists.
+  const [isSubmitted, setSubmitted] = useState<boolean>(false);
+  // Would trigger when new field button is clicked.
+  const [fieldAdd, setFieldAdd] = useState<boolean>(false);
 
   const validationSchema = Yup.object().shape({
     FieldName: Yup.string().max(24).required('Required'),
-    Area: Yup.number().min(1).max(100).required('Required'),
-    Comments: Yup.string().max(200),
+    Area: Yup.number()
+      .min(1, 'Area should be higher than 1')
+      .max(100, 'Area should be lower than 100')
+      .required('Required'),
+    Comments: Yup.string().max(200, 'Comments should be lower than 200'),
```

Comment:
'Comments should be lower than 200' chars?

Reply:
Yeah that's more clear.